packages/vite/src/node/plugins/worker.ts

```diff
@@ -174,7 +179,7 @@ export async function workerFileToUrl(
   let hash = workerMap.bundle.get(id)
   if (hash) {
     // rewrite truth id, no need to replace by asset plugin
-    return config.base + workerMap.emitted.get(hash)!
+    return config.base + workerMap.emitted.get(hash)! // TODO
```

Comment:
```suggestion
    return config.base + workerMap.emitted.get(hash)! // TODO: handle normalizing base
```

Would be nice to add what is to do here, or will it be done before this PR is merged?

Reply:
@poyoho would it be ok to replace:
```js
  let hash = workerMap.bundle.get(id)
  if (hash) {
    // rewrite truth id, no need to replace by asset plugin
    return config.base + workerMap.emitted.get(hash)! // TODO
  }
```
with:
```js
  let hash = workerMap.bundle.get(id)
  if (hash) {
    return `__VITE_ASSET__${hash}__`
  }
```
Looks like an unneeded optimization, and IIUC, it will resolve this TODO

Reply:
Maybe not, if the nested worker uses an external bundled worker but there is no resource file in the current building.

Reply:
Could you expand on this? Wouldnt this branch be hit only when a worker is used twice? Are you describing a different issue that we should also take into account that isnt related to this line?

Reply:
main.js (in vite rollup building)
```js
const worker_a = new Worker(new URL("./a.js", import.meta.url))
// to: 
const worker_a = new Worker(new URL("__VITE_ASSETS_(worker_a)__", import.meta.url))

// -------
const worker_b = new Worker(new URL("./b.js", import.meta.url))
// to:
const worker_b = new Worker(new URL("__VITE_ASSETS_(worker_b)__", import.meta.url))
```

b.js (in  worker rollup bundling)
```js
const worker_a = new Worker(new URL("./a.js", import.meta.url))
// to:
const worker_b = new Worker(new URL("__VITE_ASSETS_(worker_a)__", import.meta.url))
```

and in the asset plugin it will find the hash in 

```ts
new WeakMap<
  ResolvedConfig,
  Map<string, string>
>()
```

Because the `config` is different, so we can not find the `hash` in worker rollup bundling.

Reply:
I still don't quite understand the issue. Would you create a minimal reproduction for it?

Reply:
oh sorry, It good point. 

I've forgotten now all `emitFile` in worker is using vite rollup bundle context. And I replace all `this.emitFile` to `ctx.emitFile` fix this issues accidentally. 🤦

Reply:
So you agree with the change I proposed? 👀

Reply:
yes.

Reply:
Applied in https://github.com/vitejs/vite/pull/7644/commits/77a05e3b4fc1fc71a296509dcd75ab97c1673430, if you think we should already do this in Vite 2.9, please create a PR only including this change @poyoho

Reply:
The effect of these two writing methods is the same in 2.9.